### PR TITLE
Documenting the Batch Payment API and Batch Payment Report API

### PR DESCRIPTION
**Pull request by aengusbates**
Documenting the Batch Payment API and Batch Payment Report API.

Documenting:
- ACH and EFT. Omitting SEPA and BACS.
- Funds Transfer reports. Omitting credit card reports.

Support docs:
- https://support.na.bambora.com/bic/w/index.html#docs/batch-processing.htm

Confluence docs:
- https://confluence.beanstream.com/display/DVL/Reporting+API+Service+Guide+-+Batch+Payments+Reporting+1.0
- https://confluence.beanstream.com/display/DVL/REST+Batch+API

### CODE REVIEW

File: source/docs/references/batch_payment/index.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+## Request parameters
+
+The following parameters are passed as a JSON object on a Content-Disposition header (`Content-Disposition: form-data; name="criteria"`).
+
+- **process_now** - Overrides process_date parameter and processes transactions on receipt. Credit card transactions only.
+  - **1** – Process immediately. Any value passed in the process_date field is ignored.
+  - **0** – Transactions are processed on the date passed in the process_date field.
+- **process_date** - The date the transactions starts processing. Format **YYYYMMDD**.
+- **sub_merchant_id** - The merchant account on which to process the transactions. Required only if request is authorised by a partner account.
```
Comment (davidlight77) [inline source/docs/references/batch_payment/index.md]: authorized

File: source/docs/references/batch_payment_report/index.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+## Request
+
+The API expects an XML document in the body of the request. The XML declaration should identify the version as "1.0" and the version as "UTF-8".
+
+```
+<?xml version="1.0" encoding="UTF-8"?>
+```
+
+The API requires the XML document to include `merchantId` and `passCode` elements for authorisation, and, `serviceName` and `rptFormat` elements to identify the type and format of report to be returned. All other supported elements are filters to constrain the results in the report.
```
Comment (davidlight77) [inline source/docs/references/batch_payment_report/index.md]: authorization

File: source/docs/references/batch_payment_report/index.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+Filters are assembled as in groups containing a column name (rptFilterBy*), an operation type (rptOperationType*) and a value (rptFilterValue*). The API requires the XML document to include all three, if any one is included.
```
Comment (davidlight77) [inline source/docs/references/batch_payment_report/index.md]: Missing asterisk: rptFilterValue

File: source/docs/references/batch_payment_report/index.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+### Elements
+
+- **merchantId** - The authorising merchant ID. Numeric string. Length: 9 chars.
```
Comment (davidlight77) [inline source/docs/references/batch_payment_report/index.md]: authorizing

File: source/docs/references/batch_payment_report/index.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+- Code: "**6**", Message:	"Data validation failed"
+- Code: "**7**", Message:	"Authentication failed"
+- Code: "**8**", Message:	"Authorization failed"
+
+## Report Fields
+
+This section documents the fields returned for each of the 4 reports.
+
+### BatchPaymentsEFT
+- **merchant_id** - Authorising merchant identifier. [Numeric. 9 digits.]
```
Comment (davidlight77) [inline source/docs/references/batch_payment_report/index.md]: Authorizing

File: source/docs/guides/batch_payment/batch.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+The number of transactions that you can include in a single batch is limited by the size of the file upload. The maximum file size is 40MB. You should limit the number of transactions in a single batch request to 10,000, as you will encounter longer upload times when uploading larger files. If you need to make more calls than that, use multiple batch requests.
+
+Transactions can reference raw bank or card data, or Payment Profile IDs.
+
+### Test accounts
+
+[Test accounts](../../../forms/create_test_merchant_account) need to be configured to enable Batch Payment functionality. We enable the functionality on request. You just need to email [support.northamerica@bambora.com](mailto:support.northamerica@bambora.com) with your test account ID.
+
+## Authorising requests
```
[the remark this answers is not in the document]
Comment (aengusbates) [reply source/docs/guides/batch_payment/batch.md]: Shudder. Yep - thought you may have gone with the British spelling. I'll stick to American spelling.

File: source/docs/guides/batch_payment/batch.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+> You can generate an API key for the Batch Payment API in the [Member Area](https://web.na.bambora.com/). After logging in, select **administration**,  then **account settings**, and finally **order settings**.
+
+> On the Order Settings page, you'll find the **Batch File Upload*** section. Here you can set an API access code by clicking the **Generate New Code** button. Once you have a new code, click **Update** at the bottom of the page.
+
+
+The authorisation string is a base64 encoded concatenation of merchant_id, ":", and api_key. You Authorization header will be formatted like this.
+
+`Authorization: Passcode Base64Encoded(merchant_id:passcode)`
+
+### Authorising as an ISV
```
[the remark this answers is not in the document]
Comment (aengusbates) [reply source/docs/guides/batch_payment/batch.md]: It's an integrated software vendor. Defining long acronyms in headers messes with the display of our side navigation. Unfortunately we don't have the options of not using acronyms.

File: source/docs/guides/report/index.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+Request objects must contain a "name",  "start_date" and "end_date", "start_row" and "end_row". They can also contain "criteria" - an array of filters. Each filter object must contain a "field", an "operator" and a "value".
+
+## Response format
+
+The response body contains a JSON object. The structure of the object is dictated by the value passed for "name" property in the request ('Search' or 'TransHistoryMinimal').
+
+Note:
+1. Search and TransHistoryMinimal have a different default order.
+2. `trn_batch_no` will always be "209" for non batched transactions? Or is the batch for credit cards set according to the day? Will a credit card batch and individual credit card transactions share a batch id? What is the relation to batch file ids (10000028)?
```
[the remark this answers is not in the document]
Comment (aengusbates) [reply source/docs/guides/report/index.md]: They are questions I was unable to get to the bottom of. They are specific to credit card reports. We don't need them for the initial deploy. I'll remove them.

File: source/docs/guides/batch_payment/report.md [hunk 9]
```diff
@@ -49,15 +49,11 @@ A report request is a HTTP request using the `application/xml` content type. All
 
 ### Response
 
-The response can be returned in either JSON, XMl, TSV, CSV, or XLS formats. The content type of the response is relative to the value of the `rptFormat` parameters in the request. The
+The response can be returned in either JSON, XMl, TSV, CSV, or XLS formats. The content type of the response is relative to the value of the `rptFormat` parameters in the request. The JSON and XMl response object contains a "code" property" indicating the success of the request. This will be a number between 1 and 8, inclusive, where "1" indicates success. It also has a message property with a description of the code.
 
-JSON and XMl response object contains a "code" property" indicating the success of the request. This will be a number between 1 and 8, inclusive, where "1" indicates success. It also has a message property with a description of the code.
+A successful request returns an array of transaction/record objects. The most important properties on these  objects are `status`, `state` and `returns`.
```
Comment (davidlight77) [inline source/docs/guides/batch_payment/report.md]: does the double space show up when rendered?

**Pull request merged**